### PR TITLE
Adds proc to show variable type as string, adds this info to Varedit

### PR DESCRIPTION
**Pull request by Amunak**
Adds a proc that prints the variable type, optionally with some extra info like text/list length, types, etc. Mostly useful for debugging. In this journey I found out that there are some types that are simply "undetectable"... Like filters, procs/callables, etc.

Uses this in Varedit to make it slightly more clear which var is which. I played a lot with how it's displayed and it's kind of hard to make it visible _enough_ but not distracting or taking up too much space. In the end I opted for a simple approach that just shows the short type and keeps varedit mostly untouched otherwise:

dreamseeker_2020-11-03_02-28-24

### CODE REVIEW

**Comment by Amunak** [comment]
!review

**Comment by Amunak** [comment]
Travis pls u r killing me

**Comment by Amunak** [comment]
Alright, hopefully fixed this. It's still not very nice but at least it works properly and even the "undetectable" types can be detected and defined in the proc.

File: code/_helpers/unsorted.dm [hunk 1]
```diff
@@ -999,6 +999,90 @@ var/list/wall_items = typecacheof(list(
...
+	if(istype(V, /list))
+		return details && list_lengths ? "list([length(V)])" : "list"
+	if(istype(V, /client))
+		return "client"
+	if(istype(V, /savefile))
+		return "savefile"
+	// Finally actual objects that inherit from /datum
+	// We want to differentiate at least the basic "special" Byond types
+	var/datum/D = V
+	if(istype(V, /regex))
```
Comment (mikomyazaki) [inline code/_helpers/unsorted.dm:1037]:
```suggestion
	if(istype(D, /regex))
```

These are datums too, right?

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

File: code/_helpers/unsorted.dm [hunk 2]
```diff
@@ -999,6 +999,91 @@ var/list/wall_items = typecacheof(list(
...
+		if(V <= -INFINITY)
+			return details ? "float(-INF)" : "float"
+		return details && print_numbers ? "float([V])" : "float"
+	// Resource types
+	if(isicon(V))
+		return "icon"
+	if(isfile(V))
+		return "file"
+	// Types that don't inherit from /datum (note that /world is not here because you can't hold a reference to it)
+	if(istype(V, /list))
```
Comment (NonQueueingMatt) [inline code/_helpers/unsorted.dm:1029]: islist is a thing now:
```suggestion
	if(islist(V))
```

File: code/_helpers/unsorted.dm [hunk 3]
```diff
@@ -999,6 +999,91 @@ var/list/wall_items = typecacheof(list(
...
+		return details && list_lengths ? "list([length(V)])" : "list"
+	if(istype(V, /client))
```
Comment (NonQueueingMatt) [inline code/_helpers/unsorted.dm:1031]:
```suggestion
	if(isclient(V))
```

File: code/_helpers/unsorted.dm [hunk 4]
```diff
@@ -999,6 +999,91 @@ var/list/wall_items = typecacheof(list(
...
+	if(isdatum(D))
+		return details ? "datum([D.type])" : "datum"
+	if(istype(D)) // let's future proof ourselves
+		return details ? "unknown-object([D.type])" : "unknown-object"
+	// some undetectable types
+	var/refType = get_type_ref_bytes(V)
+	if(refType == "")
+		return "unknown"
+	if(refType == get_type_ref_bytes(known_proc)) // it's a proc of some kind
+		if(istext(V?:name) && V:name != "") // procs with names are generally verbs
```
Comment (NonQueueingMatt) [inline code/_helpers/unsorted.dm:1075]: Is `name` not null on procs with no `set name` specified? Huh.

**Review by NonQueueingMatt — APPROVED** [review]

File: code/_helpers/unsorted.dm (hunk 2 above)
Comment (NonQueueingMatt) [reply code/_helpers/unsorted.dm:1029]: This would also make special lists like contents/overlays show as list, but idk if that's something you want

Comment (Amunak) [reply code/_helpers/unsorted.dm:1029]: I'm not sure actually, like it'd be better than showing as unknown, but also should probably be differentiated.

What I'll probably do is keep this as a `istype` check and then do `islist` just before the end.

File: code/_helpers/unsorted.dm (hunk 4 above)
Comment (Amunak) [reply code/_helpers/unsorted.dm:1075]: No idea, trying to cover my bases. For all intents and purposes a verb without a name can't be used as a proc. So even if someone defined it as verb and/or changed the name in initialize or something to empty string it should show up as a proc.

File: code/_helpers/unsorted.dm (hunk 2 above)
Comment (Amunak) [reply code/_helpers/unsorted.dm:1029]: Well I did some experimenting and I can't find anything that would register in `islist` and not with `istype list`, so I'll just use the Byond proc.

**Review by Arrow768 — APPROVED** [review]

**Review by mikomyazaki — APPROVED** [review]

**Pull request merged**